ORK1Kit/ORK1Kit/Common/ORK1NavigableOrderedTask.m

```diff
@@ -174,7 +180,7 @@ - (ORK1Step *)stepAfterStep:(ORK1Step *)step withResult:(ORK1TaskResult *)result
         
         ORK1SkipStepNavigationRule *skipNavigationRule = _skipStepNavigationRules[nextStep.identifier];
         if ([skipNavigationRule stepShouldSkipWithTaskResult:result]) {
-            return [self stepAfterStep:nextStep withResult:result];
+            return [self stepAfterStep:nextStep withResult:result ignoreSaveOnArrival:YES];
```

Comment:
Checking my understanding:

- Given steps A, B, and C, where B.saveOnArrival = YES
- There's a skip navigation rule that indicates step B should be skipped
- `stepAfterStep:A` is called

At this point in the code, `nextStep` has been set to `B`, but we need to find the step after B since it's being skipped. Thus we call `[stepAfterStep:B ignoreSaveOnArrival:YES]`, which determines that C should be the next step?

Reply:
That is correct. In early testing of the prototype, I noticed it would auto-complete (advance and return nil) if there were a skip rule on the last step. This is a way to pass on a little state recursively to handle that case.